app/src/main/java/com/github/se/eventradar/ui/login/SignUpScreen.kt

```diff
@@ -0,0 +1,469 @@
+                    "Birth Date" to birthDate.value,
+                    "Email" to user?.email,
+                    "Profile Picture" to selectedImageUri.value.toString()
+                )
+
+                // Add a new document with a generated ID into collection "users"
+                Firebase.firestore.collection("users").add(userValues)
+                    .addOnSuccessListener { documentReference ->
+                        Log.d(TAG, "DocumentSnapshot added with ID: ${documentReference.id}")
+                        Toast.makeText(
+                            context,
+                            "User data added successfully",
+                            Toast.LENGTH_SHORT
+                        ).show()
+                    }.addOnFailureListener { e ->
+                        Log.w(TAG, "Error adding document", e)
+                        Toast.makeText(context, "Error adding user data", Toast.LENGTH_SHORT)
+                            .show()
+                    }
+                 */
+
+                navigationActions.navController.navigate(Route.OVERVIEW)
+
+            } else {
+                // Handle the error
+                openErrorDialog.value = true
+            }
+        }
+    )
+
+    ErrorDialogBox(openErrorDialog = openErrorDialog)
+
+    val imagePickerLauncher =
+        rememberLauncherForActivityResult(contract = ActivityResultContracts.GetContent(),
+            onResult = { uri ->
+                // Handle the returned Uri
+                selectedImageUri.value = uri
+            })
+
+    val providers = arrayListOf(AuthUI.IdpConfig.GoogleBuilder().build())
+    val intent = AuthUI.getInstance().createSignInIntentBuilder().setIsSmartLockEnabled(false)
+        .setAvailableProviders(providers).build()
+
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .testTag("signUpScreen"),
+        verticalArrangement = Arrangement.Center,
+        horizontalAlignment = Alignment.CenterHorizontally,
+    ) {
+        ConstraintLayout {
+            val (titleRow, profilePicture, usernameField, nameField, surnameField, phoneField, birthDateField, signInButton) = createRefs()
+            Row(
+                modifier = Modifier.constrainAs(ref = titleRow) {
+                    top.linkTo(
+                        parent.top,
+                        margin = 16.dp
+                    )
+                    centerHorizontallyTo(parent)
+                },
+                horizontalArrangement = Arrangement.Center,
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                Image(
+                    painter = painterResource(R.drawable.event_radar_logo),
+                    contentDescription = "Logo",
+                    modifier = Modifier
+                        .padding(1.dp)
+                        .width(300.dp)
+                        .height(100.dp)
+                        .testTag("logo"),
+                )
```

Comment:
Important: Upon running it in my machine, the Sign In with Google Button gets cut off from the UI.
nit: I think reducing the size of the Logo would be helpful.

Reply:
That is definitely weird... will look into it. On my side I get this 
![image](https://github.com/swent-sp-2024-event-radar/event-radar-app/assets/100644643/f7782bb4-006f-480e-808c-1d1e674c575c)

Reply:
<img width="196" alt="Screen Shot 2024-04-12 at 18 49 08" src="https://github.com/swent-sp-2024-event-radar/event-radar-app/assets/54261454/11a4c363-34fc-440c-9b24-65f61196e423">
I've pulled it again and it looks good, I think I was working off an old commit, sorry! (but I meant the Log in here was cutoff, but looks ok now)